src/main/java/bio/terra/cda/app/service/QueryService.java

```diff
@@ -79,12 +80,19 @@ private Results getJobResults(Job queryJob) {
     try {
       // Get the results.
       TableResult result = queryJob.getQueryResults();
+      FieldList fields = result.getSchema().getFields();
 
       Results results = new Results();
 
-      // Copy all row data to results. Each row is a JSON object.
+      // Copy all row data to results. For each row, collect the fields as JSON key/value pairs.
       for (FieldValueList row : result.iterateAll()) {
-        var rowData = row.get(0).getStringValue();
+        List<String> fieldData = new ArrayList<>();
+        for (int i = 0; i < fields.size(); i++) {
+          fieldData.add(
+              String.format(
+                  "\"%s\":\"%s\"\n", fields.get(i).getName(), row.get(i).getStringValue()));
+        }
+        String rowData = "{" + String.join(",", fieldData) + "}";
```

Comment:
Would it be safer to use something like Jackson to serialize the Json?  You'd save yourself from potential encoding issues.

Reply:
That was a great idea, by changing this to use a json object I was able to remove the RawJsonValue hack I added in a while back.

Reply:
That was a great idea, by changing this to use a json object I was able to remove the RawJsonValue hack I added in a while back.